integration/tests/git_source/git_source_repeated_remote.rb

```diff
@@ -0,0 +1,65 @@
+require 'git_utils'
+require 'r10k_utils'
+require 'master_manipulator'
+test_name 'Verify the same remote can be used in more than one object'
+
+env_path = on(master, puppet('config print environmentpath')).stdout.rstrip
+r10k_fqp = get_r10k_fqp(master)
+git_environments_path = '/root/environments'
+git_repo_path = '/git_repos'
+git_repo_name = 'environments'
+git_control_remote = File.join(git_repo_path, "#{git_repo_name}.git")
+code_dir = '/etc/puppetlabs/code/environments/production'
+
+last_commit = git_last_commit(master, git_environments_path)
+git_provider = ENV['GIT_PROVIDER']
+r10k_config_path = get_r10k_config_file_path(master)
+r10k_config_bak_path = "#{r10k_config_path}.bak"
+#In-line files
+r10k_conf = <<-CONF
+cachedir: '/var/cache/r10k'
+git:
+  provider: '#{git_provider}'
+sources:
+  control:
+    basedir: "#{env_path}"
+    remote: "#{git_control_remote}"
+CONF
+
+puppetfile = <<-EOS
+mod 'non_module_object_1',
+  :install_path => './',
+  :git => 'git://github.com/puppetlabs/control-repo.git',
+  :branch => 'production'
+
+mod 'non_module_object_2',
+  :install_path => '',
```

Comment:
what is the behavior when `:install_path` is an empty string?
I'm assuming `:install_path` is where to install the module and having a different path than the one for `non_module_object_1` is how we ensure these are treated as separate modules?
Could they not just go in `./mod1` and `./mod2` or something?

Reply:
These are specifically not modules. 

And yeah I imagine the install path could be anything. This test is copied almost exactly from the other one that failed, minus the assertions for purging. I didn't dig much into the choices made there (the commits for these test are notoriously bare-bones, and their tickets tend to be in a different tracking system than JIRA, so the spelunking is difficult), but I can.

Reply:
I don't mind you leaving it, if that's the existing pattern we've established. just trying to understand how all things things work.

if they're not modules, what are they?

Reply:
Also, `install_path` appears to be relative to the location of the Puppetfile. So this is equivalent to the `./` in the previous example, which was likely intentional in the original test, which was verifying that things at the top level (i.e. not in the module dir of an environment) would get purged.

Reply:
In this specific test, they're whole control repos. But according to the docs (linked in the other comment) they're most commonly hiera data in a real install.

Reply:
so would 'content that isn't a forge module' be more accurate? like, this could be anything that lives in a git repo, right? even what we call a "module"? sorry, this is just background stuff that I'm trying to understand.

Reply:
I'm going to change this to not muddy the waters in this test. But yeah, in this case, it's something that isn't structured like a Puppet module, which was a specific feature that got added. Previously, things had to look like modules for r10k to deploy them (I gather from skimming the ticket), whether they came from the Forge or from git. But now, the thing on the other end of a git source can be something else besides a Puppet module if the user wants.